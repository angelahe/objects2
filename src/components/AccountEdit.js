import React from 'react';
import '../styles/Styles140.css'
import editbtn from '../images/edit_FFFFFF.png'

const AccountEdit = (props) => {

//    console.log("in accountedit props is", props);
//    console.log("selected account is ", props.index);
//    console.log("account name is ", props.AccountList.Accounts[props.index].acctName);

    function handleUpdateClick() {
        const newName = document.getElementById("acctName").value;
        const newBalance = Number(document.getElementById("acctBalance").value);
        props.updateAccount(newName, newBalance, props.account);
    }

    const accountName = props.account.acctName;
    const accountBalance = props.account.balance;

    return(
        <div>
            <div className = "ItemBox AppHeader">
                <span className = "AddItem">Edit Account</span>
                <button className="AppBtn "><img className="btnImg" src={editbtn} alt="Add"/></button>
            </div>
            <div className = "ItemBox">
                <span className = "DetailText">Account Name:</span>
                <input className="InputText" id="acctName" defaultValue={accountName} autoFocus={true}></input><br/>
                <span className = "DetailText">Balance:</span>
                <input className = "InputText" id = "acctBalance" type="number" defaultValue={accountBalance}></input><br/><br/>
                <button className="btnText" onClick={handleUpdateClick}>Update Account</button>

            </div>
        </div>
    )

}
export default AccountEdit;