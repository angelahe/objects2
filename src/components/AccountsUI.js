import React from 'react';
import Accounts from './accounts';
import Account from './account';
import AccountSummary from './AccountSummary';
import AccountCreate from './AccountCreate';
import AccountEdit from './AccountEdit';
import AccountDelete from './AccountDelete';
import AccountListItem from './AccountListItem';
import AccountActions from './AccountActions';
import addbtn from '../images/add_FFFFFF.png';

import '../styles/Styles140.css';


class AccountsUI extends React.Component {

    constructor(props) {
        super(props)
        this.state = {
            AccountList: new Accounts(),
            nextId: 0,
            currentAccount: null,
            addAccount: false,
            editShow: false,
            deleteShow: false,
            accountActionsShow: false
        }
    }

    onBtnAddClick = () => {
        const changeState = !this.state.addAccount;
        this.setState ({addAccount: changeState});
    }

    //make a copy of the current AccountList in state
    makeAccountCopy = () => {
        const AccountListCopy = new Accounts();
        for(let i=0; i<this.state.AccountList.Accounts.length; i++) {
            AccountListCopy.addAccount(this.state.AccountList.Accounts[i]);
        }
        return AccountListCopy;

    }

    onAccountCreate = (name, balance) => {
        const AccountListCopy = this.makeAccountCopy();
        AccountListCopy.addAccount(new Account(balance, name, this.state.nextId));

        this.setState({
            AccountList: AccountListCopy,
            addAccount: !this.state.addAccount,
            nextId: this.state.nextId + 1
        });
    }

    onEditClick = (account) => {
        this.setState({
            editShow: !this.state.editShow,
            currentAccount: account
        });

    }

    //only update state if the accountupdate was successful
    onAccountUpdate = (newName, newBalance, account) => {
        this.state.AccountList.updateAccount(newName, newBalance, account)

        this.setState({
                AccountList: this.state.AccountList,
                editShow: !this.state.editShow
            });

    }

    onDeleteClick = (account) => {
        this.setState({
            deleteShow: true,
            currentAccount: account
        });
    }

    onDeleteAccount = (account) => {
        this.state.AccountList.deleteAccount(account);

        this.setState({
            AccountList: this.state.AccountList,
            deleteShow: false,
            currentAccount: this.state.AccountList[0]
        });
    }

    onAccountClick = (account) => {
        this.setState({
            accountActionsShow: true,
            currentAccount: account
        })
    }

    onDepositToAccount = (amount, account) => {
        this.state.AccountList.depositToAccount(account, amount);

        this.setState({
            AccountList: this.state.AccountList,
            accountActionsShow: false,
        });
    }

    onWithdrawFromAccount = (amount, account) => {
        this.state.AccountList.withdrawFromAccount(account, amount);

        this.setState({
            AccountList: this.state.AccountList,
            accountActionsShow: false
        });
    }

    render() {

        console.log("addAccount is ", this.state.addAccount);
        const accountListItems = this.state.AccountList.Accounts.map((item, index) =>
           <AccountListItem key={item.acctId}
                            index = {index}
                            account={item}
                            accountClicked={this.onAccountClick}
                            editClicked={this.onEditClick}
                            deleteClicked={this.onDeleteClick}
           />
        );

        return(
            <div>
                <h1>Hello from Accounts</h1>
                <div className = "AppContainer">
                    <AccountSummary accounts={this.state.AccountList} />
                    <div className = "AppPanel">
                          <div className = "ItemBox AppHeader">
                            <span className = "AddItem">Add Account</span>
                            <button className="AppBtn" onClick={this.onBtnAddClick}>
                                <img className="btnImg" src={addbtn} alt="Add"/>
            `                    </button>
                          </div>
                        <div className = "AppList">
                            {accountListItems}
                        </div>
                    </div>
                    <div className = "AppPanel">
                        {(this.state.addAccount)
                            ? <AccountCreate {...this.state}
                                             createClicked = {this.onAccountCreate}
                              />
                            : null }
                        {(this.state.editShow)
                            ? <AccountEdit
                                         account={this.state.currentAccount}
                                         updateAccount = {this.onAccountUpdate}
                              />
                            :null }
                        {(this.state.deleteShow)
                            ? <AccountDelete
                                           account={this.state.currentAccount}
                                           deleteAccount = {this.onDeleteAccount}
                              />
                            :null }
                        {(this.state.accountActionsShow)
                            ? <AccountActions
                                              account={this.state.currentAccount}
                                              deposit={this.onDepositToAccount}
                                              withdraw={this.onWithdrawFromAccount}
                            />
                            : null
                        }
                    </div>

                </div>

            </div>
        )

    }
}
export default AccountsUI