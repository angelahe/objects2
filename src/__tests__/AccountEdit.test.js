import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import AccountEdit from "../components/AccountEdit";
import AccountsUI from "../components/AccountsUI";
import Account from "../components/account";
import Accounts from "../components/accounts";

import renderer from "react-test-renderer";

let props = new AccountsUI();

const newAccount = new Account(100, "Test", 0);
const newAccounts = new Accounts();
newAccounts.addAccount(newAccount);
//console.log("newAccounts is ", newAccounts);
props.state.AccountList = newAccounts;
const account = newAccount;

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<AccountEdit account={account}
                               updateAccount = {props.onAccountUpdate}
                  />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders without crashing', () => {
  shallow(<AccountEdit account={account}
                       updateAccount = {props.onAccountUpdate}
          />);
});

test('AccountUI renders correctly', () => {
  const tree = renderer.create(<AccountEdit account={account}
                                            updateAccount = {props.onAccountUpdate}
                               />).toJSON();
  expect(tree).toMatchSnapshot();
});