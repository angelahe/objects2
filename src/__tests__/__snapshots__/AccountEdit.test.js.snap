// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountUI renders correctly 1`] = `
<div>
  <div
    className="ItemBox AppHeader"
  >
    <span
      className="AddItem"
    >
      Edit Account
    </span>
    <button
      className="AppBtn "
    >
      <img
        alt="Add"
        className="btnImg"
        src="edit_FFFFFF.png"
      />
    </button>
  </div>
  <div
    className="ItemBox"
  >
    <span
      className="DetailText"
    >
      Account Name:
    </span>
    <input
      autoFocus={true}
      className="InputText"
      defaultValue="Test"
      id="acctName"
    />
    <br />
    <span
      className="DetailText"
    >
      Balance:
    </span>
    <input
      className="InputText"
      defaultValue={100}
      id="acctBalance"
      type="number"
    />
    <br />
    <br />
    <button
      className="btnText"
      onClick={[Function]}
    >
      Update Account
    </button>
  </div>
</div>
`;
